internal/errors/errors.go

```diff
@@ -0,0 +1,170 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package errors defines the error handling used by Kpt codebase.
+package errors
+
+import (
+	"fmt"
+	"strings"
+)
+
+// Error is an implementation of the error interface used in the kpt
+// codebase.
+// It is based on the design in https://commandcenter.blogspot.com/2017/12/error-handling-in-upspin.html
+type Error struct {
```

Comment:
Maybe comment about the intended usage of this abstraction. Will this be shown the end-user of the CLI? Is it to facilitate better error handling/recovery as an implementation detail?

Reply:
Added some more context about the intention here. I would have liked to be more concrete here, but we haven't formed the metal model fully yet.